### PR TITLE
Handle fndef rendering together with signature rendering

### PR DESCRIPTION
**Pull request by oli-obk**
Pulled out of https://github.com/rust-lang/rust/pull/134353

Changes some highlighting in type mismatch errors around fndefs

### CODE REVIEW

File: compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs [hunk 1]
```diff
@@ -1318,36 +1339,21 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
...
-                values.0.push(path1, !same_path);
-                values.1.push(path2, !same_path);
-                values
+                self.cmp_fn_sig(&sig1, Some((*did1, args1)), &sig2, Some((*did2, args2)))
             }
 
             (ty::FnDef(did1, args1), ty::FnPtr(sig_tys2, hdr2)) => {
                 let sig1 = self.tcx.fn_sig(*did1).instantiate(self.tcx, args1);
-                let mut values = self.cmp_fn_sig(&sig1, &sig_tys2.with(*hdr2));
-                values.0.push_highlighted(format!(
```
Comment (oli-obk) [inline compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:1333]: Previously only the fn path on the lhs was highlighted, but not on the rhs (the arm below uses `push_normal`)

**Review by compiler-errors — APPROVED** [review]

**Comment by matthiaskrgr** [comment]
@bors r-

https://github.com/rust-lang/rust/pull/134360#issuecomment-2544413822

**Comment by oli-obk** [comment]
blessed windows output, too

@bors r=compiler-errors

**Pull request merged**